### ISSUE TITLE
Merge error for a mixture of loaded and lazy data

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#2073]
I've stumbled across a problem whereby cubes would not merge, and eventually realized it was because some of the data has been loaded but some hadn't.  Here's an example that reproduces the error:

```python
filename = iris.sample_data_path('uk_hires.pp')
cubes =  iris.load_raw(filename)[1:3]
a = cubes[0].data
print cubes.merge_cube()
```
the merge fails with this error:
```
    raise iris.exceptions.MergeError(msgs)
iris.exceptions.MergeError: failed to merge into a single cube.
  cube data fill_value differs: 1e+20 != -1e+30
```
In this example the cubes will obviously merge fine if they are either both loaded or both lazy.  I doubt it makes sense to try to fix this so that they merge in the case where only some data is loaded (presumably you'd have to load everything?)  However, it might be helpful to have a specific error message for the loaded/lazy case, to make a work-around more obvious to the user.

**Comment by ajdawson** [comment]
After #2075 fill value is simply ignored on merge, fixing this issue.
